src/benchmarks/micro/corefx/System.Text.Json/Utf8JsonWriter/Perf.Large.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace System.Text.Json
+{
+    [BenchmarkCategory(Categories.CoreFX, Categories.JSON)]
+    public class Perf_Large
+    {
+        private static readonly byte[] ExtraArray = Encoding.UTF8.GetBytes("ExtraArray");
+        private static readonly byte[] First = Encoding.UTF8.GetBytes("first");
+        private static readonly byte[] Last = Encoding.UTF8.GetBytes("last");
+        private static readonly byte[] Age = Encoding.UTF8.GetBytes("age");
+        private static readonly byte[] PhoneNumbers = Encoding.UTF8.GetBytes("phoneNumbers");
+        private static readonly byte[] Address = Encoding.UTF8.GetBytes("address");
+        private static readonly byte[] Street = Encoding.UTF8.GetBytes("street");
+        private static readonly byte[] City = Encoding.UTF8.GetBytes("city");
+        private static readonly byte[] Zip = Encoding.UTF8.GetBytes("zip");
+
+        private const int DataSize = 100_000;
+
+        private ArrayBufferWriter<byte> _arrayBufferWriter;
+        private PooledBufferWriter<byte> _pooledBufferWriter;
+
+        private string[] _propertNames;
+        private byte[][] _propertNamesUtf8;
+        private int[] _numberArrayValues;
+        private string[] _stringArrayValues;
+
+        [Params(true, false)]
+        public bool Formatted;
+
+        [Params(true, false)]
+        public bool SkipValidation;
+
+        [Params(true, false)]
+        public bool Pool;
+
+        [GlobalSetup]
+        public void Setup()
+        {
+            _arrayBufferWriter = new ArrayBufferWriter<byte>();
+            _pooledBufferWriter = new PooledBufferWriter<byte>();
```

Comment:
Since this is never disposed your perf test is just stealing from the array pool(s).

It also seems like this is really testing the perf of those IBufferWriter implementations, and thus probably not appropriate to this suite.  I'd get rid of the Pool boolean altogether.

Reply:
Since the JsonWriter is the primary consumer of this interface, I saw value in adding perf tests for various implementations of the interface (works as integration tests for a more end-to-end, "real", scenario). But, I am fine with removing it here and only measuring IBufferWriter implementation in isolation, separately (though would seem "fake" to me).

Reply:
"fake" is fine. These microbenchmarks should not be trying to encompass "entirely real use of this type". They should be hyperfocused to isolate parts of the type to highlight small regressions. The smaller the code in a `[Benchmark]` is the better.